corehq/apps/api/es.py

```diff
@@ -681,6 +478,21 @@ def consume_params(self, raw_params):
             return filters.term(self.term, value)
 
 
+class XFormServerModifiedParams:
```

Comment:
for form queries we should always use this one instead of the one defined below `DateRangeParams('server_modified_on'),`

you could change `query_param_consumers` into a function that returns the list based on the doc_type:

```python
default_query_param_consumers = [...]

def get_query_consumers(doc_type):
     query_consumers = list(default_query_param_consumers)
     if doc_type == 'form':
        query_consumers.append(XFormServerModifiedParams())
     elif doc_type == 'case':
        query_consumers.append(DateRangeParams('server_modified_on'))
    return query_consumers
```

Reply:
Will update

Reply:
Though, `DateRangeParams('server_modified_on')` could still be used for forms outside of data export tool, no? XFormServerModifiedParams being a complete custom ES filter from GET params.